tutorials/xr/basic_xr_locomotion.rst

```diff
@@ -3,5 +3,65 @@
+
+Adding a floor
+--------------
+
+This node governs the in game movement of your character and will immediately react to gravity. So to prevent our player from infinitely falling down we'll quickly add a floor to our scene.
+
+We start by adding a :ref:`StaticBody3D <class_staticbody3d>` node to our root node and we rename this to ``Floor``.
+We add a :ref:`MeshInstance3D <class_meshinstance3d>` node as a child node for our ``Floor`` and create a new :ref:`PlaneMesh <class_planemesh>` as it's mesh. For now we set the size of the mesh to 100 x 100 meters.
+Next we add a :ref:`CollisionShape3D <class_collisionshape3d>` node as a child node for our ``Floor`` and create a ``BoxShape`` as our shape. We set the size of this box shape to 100 x 1 x 100 meters. We also need to move our collision shape down by 0.5 meters so the top of our box is flush with the floor.
+
+To make it easier to see that we're actually moving around our world, a white floor isn't going to do it. For this we add a material to our PlaneMesh and set the albedo to a grid texture we've created with `Wahooneys excellent free texture generator <https://wahooney.itch.io/texture-grid-generator>`_. 
+
+.. image:: img/godot_xr_tools_floor.webp
+
+Direct movement
+---------------
+
+We're going to start adding some basic direct movement to our setup. This allows the user to move through the virtual world using joystick input.
+
+.. note::
+  It is important to note that moving through the virtual world while the player is standing still in the real world can be nausia inducing especially for players who are new to VR. The default settings on our movement functions are fairly conservative. We advise you to stick to these defaults but offer features in game to enable less comfortable settings for more experienced users who are used to playing VR games.
+
+We want to enable this on the right hand controller and we do this by adding a subscene to the right hand :ref:`XRController3D <class_xrcontroller3d>` node. Select ``addons/godot-xr-tools/functions/movement_direct.tscn`` as the scene to add.
+
+This function adds forward and backwards movement to the player by using the joystick on the right hand controller. It has an option to also add left/right strafe but by default this is disabled.
+
+Instead we are going to add the ability for the player to also turn with this joystick so we will add another subscene to our controller node, select ``addons/godot-xr-tools/functions/movement_turn.tscn`` for this.
+
+The turn system by default uses a snap turn approach. This means that turning happens in steps. This may seem jarring however it is a tried and tested method of combatting motion sickness. You can easily switch to a mode that offers smooth turning by changing the ``mode`` property on the turn node.
```

Comment:
```suggestion
The turn system by default uses a snap turn approach. This means that turning happens in steps. This may seem jarring however it is a tried and tested method of combating motion sickness. You can easily switch to a mode that offers smooth turning by changing the ``mode`` property on the turn node.
```